pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -577,15 +579,30 @@ private TimeSeriesDispatchClient getOrCreateTimeSeriesDispatchClient(
     return _timeSeriesDispatchClientMap.computeIfAbsent(key, k -> new TimeSeriesDispatchClient(hostname, port));
   }
 
+  private static QueryResult runReducerFromQueryThread(
```

Comment:
The name of this method is confusing, it's running on the caller thread right? Should it be something like `runReducerFromQueryThreadContext` instead?